File: api/Models/Enums.cs
namespace BoatAdminApi.Enums
{
    public enum ListingType
    {
        Dealer,
        Broker,
        Concessionnaire,
        Private
    }

    public enum BoatStatus
    {
        InStock = 1,
        ComingSoon,
        DealPending,
        Incoming,
        IncomingDealPending,
        OnOrder,
        OutOfStock,
        Sold,
        Unavailable,
        UnderDeposit
    }

    public enum Condition
    {
        New,
        Used
    }

    public enum DriveType
    {
        NoEngine,
        InboardOutboard,
        Inboard,
        Outboard,
        Jet,
        VDrive,
        Other
    }

    public enum FuelType
    {
        None,
        Gas,
        Diesel,
        Electric
    }
}

File: client/src/enums/enums.ts
export const ListingType = ["Dealer", "Broker"] as const;
export type ListingType = (typeof ListingType)[number];

export const BoatStatus = [
  "InStock",
  "ComingSoon",
  "DealPending",
  "Incoming",
  "IncomingDealPending",
  "OnOrder",
  "OutOfStock",
  "Sold",
  "Unavailable",
  "UnderDeposit",
] as const;
export type BoatStatus = (typeof BoatStatus)[number];

export const Condition = ["New", "Used"] as const;
export type Condition = (typeof Condition)[number];

export const DriveType = [
  "NoEngine",
  "InboardOutboard",
  "Inboard",
  "Outboard",
  "Jet",
  "VDrive",
  "Other",
] as const;
export type DriveType = (typeof DriveType)[number];

export const FuelType = ["None", "Gas", "Diesel", "Electric"] as const;
export type FuelType = (typeof FuelType)[number];
